src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -726,49 +747,71 @@ ParameterProvider getParameterProvider() {
     return parameterProvider;
   }
 
-  /***
+  /**
    * Registers the performance metrics along with JVM memory and Threads.
    *
-   * Latency and throughput metrics are emitted to JMX, jvm memory and thread metrics are logged to Slf4JLogger
+   * <p>Latency and throughput metrics are emitted to JMX, jvm memory and thread metrics are logged
+   * to Slf4JLogger
    */
-  private void registerMetricsForClient() {
+  private void setupMetricsForClient() {
+    // Start the telemetry background worker if needed
+    if (ENABLE_TELEMETRY_TO_SF) {
+      this.telemetryWorker = Executors.newSingleThreadScheduledExecutor();
+      this.telemetryWorker.scheduleWithFixedDelay(
+          this::reportStreamingIngestTelemetryToSF,
+          STREAMING_INGEST_TELEMETRY_UPLOAD_INTERVAL_IN_SEC,
+          STREAMING_INGEST_TELEMETRY_UPLOAD_INTERVAL_IN_SEC,
+          TimeUnit.SECONDS);
+    }
+
+    // Register metrics if needed
     metrics = new MetricRegistry();
-    // Blob histograms
-    blobSizeHistogram = metrics.histogram(MetricRegistry.name("blob", "size", "histogram"));
-    blobRowCountHistogram =
-        metrics.histogram(MetricRegistry.name("blob", "row", "count", "histogram"));
-    cpuHistogram = metrics.histogram(MetricRegistry.name("cpu", "usage", "histogram"));
-
-    // latency metrics
-    flushLatency = metrics.timer(MetricRegistry.name("latency", "flush"));
-    buildLatency = metrics.timer(MetricRegistry.name("latency", "build"));
-    uploadLatency = metrics.timer(MetricRegistry.name("latency", "upload"));
-    registerLatency = metrics.timer(MetricRegistry.name("latency", "register"));
-
-    // throughput metrics
-    uploadThroughput = metrics.meter(MetricRegistry.name("throughput", "upload"));
-    inputThroughput = metrics.meter(MetricRegistry.name("throughput", "input"));
-    SharedMetricRegistries.add(SNOWPIPE_STREAMING_SHARED_METRICS_REGISTRY, metrics);
-
-    JmxReporter jmxReporter =
-        JmxReporter.forRegistry(this.metrics)
-            .inDomain(SNOWPIPE_STREAMING_JMX_METRIC_PREFIX)
-            .convertDurationsTo(TimeUnit.SECONDS)
-            .createsObjectNamesWith(
-                (ignoreMeterType, jmxDomain, metricName) ->
-                    getObjectName(this.getName(), jmxDomain, metricName))
-            .build();
-    jmxReporter.start();
-
-    // add JVM and thread metrics too
-    jvmMemoryAndThreadMetrics = new MetricRegistry();
-    jvmMemoryAndThreadMetrics.register(
-        MetricRegistry.name("jvm", "memory"), new MemoryUsageGaugeSet());
-    jvmMemoryAndThreadMetrics.register(
-        MetricRegistry.name("jvm", "threads"), new ThreadStatesGaugeSet());
-
-    SharedMetricRegistries.add(
-        SNOWPIPE_STREAMING_JVM_MEMORY_AND_THREAD_METRICS_REGISTRY, jvmMemoryAndThreadMetrics);
+
+    if (ENABLE_TELEMETRY_TO_SF || this.parameterProvider.hasEnabledSnowpipeStreamingMetrics()) {
```

Comment:
May be we can remove `this.parameterProvider.hasEnabledSnowpipeStreamingMetrics()`? since ENABLE_TELEMETRY_TO_SF is always true!

Reply:
Having the ENABLE_TELEMETRY_TO_SF constant is to prevent the case when the telemetry change has some issues, so we can quickly create a patch release with ENABLE_TELEMETRY_TO_SF set to false